crates/sui-indexer/src/store/pg_indexer_store.rs

```diff
@@ -807,27 +820,76 @@ impl<T: R2D2Connection + 'static> PgIndexerStore<T> {
                 tracing::error!("Failed to persist tx_changed_objects with error: {}", e);
             })
         }));
+
+        futures.push(self.spawn_blocking_task(move |this| {
```

Comment:
(Optional) It seems like you should be able to abstract this away maybe? Seems like we're repeating it enough for that to be worthwhile.

Reply:
that's fair, i was considering it too. just wasnt sure to introduce it here or at the end. i guess it doesn't really matter

Reply:
I'll revisit this after landing the changes needed to start backfilling